### PR TITLE
Improve integration tests

### LINKED ISSUE #69: Improve integration tests
### Need
```gherkin
As a developer
I want to use temporary files
When testing a package
```

### Solution
Use [tmp](https://www.npmjs.com/package/tmp).
Test the package with a recursive rootDirectory.

### PR DESCRIPTION
**Pull request by adela-istrate-hs**
Closes #69

### CODE REVIEW

**Comment by adela-istrate-hs** [comment]
@valentin-radulescu-hs  @andrei-picus-hs  Can you review?

File: tests/integration/test.js [hunk 1]
```diff
@@ -146,28 +153,19 @@ describe('Mugshot integration', function() {
...
     var mugshotOptions = {
       acceptFirstBaseline: false
     };
 
-    return wdioInstance = wdio.remote(options).init().url(url).then(function() {
-      var browser = new WdioAdapter(this);
-      mugshot = new Mugshot(browser, mugshotOptions);
-    });
+    return wdioInstance =
+        wdio.remote(BROWSER_OPTIONS).init().url(URL).then(function() {
```
Comment (andrei-picus-hs) [inline tests/integration/test.js]: Break on a property access (preferably `.then`).

File: tests/integration/test.js [hunk 2]
```diff
@@ -190,3 +188,51 @@ describe('Mugshot integration with disabled acceptFirstBaseline', function() {
...
+                    noDifferencesSelector.name + ext)
+              },
+              mugshot = new Mugshot(browser, mugshotOptions);
+
+          mugshot.test(noDifferencesSelector, function(error, result) {
+            expect(error).to.be.null;
+            expect(result).to.be.deep.equal(expected);
+
+            // Manual cleanup
+            cleanupCallback();
```
Comment (andrei-picus-hs) [inline tests/integration/test.js]: Any reason for this? The docs say `tmp` will cleanup after itself.

File: tests/integration/test.js [hunk 3]
```diff
@@ -59,20 +46,40 @@ describe('Mugshot integration', function() {
...
         diff: paths[2]
-      },
-      wdioInstance, mugshot;
+      };
 
-  before(function() {
-    var options = {
-      desiredCapabilities: {
-        browserName: 'phantomjs'
+function cleanUp() {
```
Comment (andrei-picus-hs) [inline tests/integration/test.js]: Do we still need this? Can we replace it with `tmp`?

File: tests/integration/test.js [hunk 4]
```diff
@@ -59,20 +46,40 @@ describe('Mugshot integration', function() {
...
+  }
+}
+
+describe('Mugshot integration', function() {
+  this.timeout(0);
+
+  var wdioInstance, mugshot;
+
+  before(function() {
+
```
Comment (andrei-picus-hs) [inline tests/integration/test.js]: Please remove empty lines like this.

File: tests/integration/test.js [hunk 5]
```diff
@@ -59,20 +46,40 @@ describe('Mugshot integration', function() {
...
+    return wdioInstance =
+        wdio.remote(BROWSER_OPTIONS).init().url(URL).then(function() {
```
Comment (andrei-picus-hs) [inline tests/integration/test.js]: See https://github.com/uberVU/mugshot/pull/70/files#r77531702.

File: tests/integration/test.js [hunk 6]
```diff
@@ -190,3 +188,51 @@ describe('Mugshot integration with disabled acceptFirstBaseline', function() {
     return wdioInstance.end();
   });
 });
+
+
+describe('Mugshot integration with recursive rootDirectory', function() {
```
Comment (andrei-picus-hs) [inline tests/integration/test.js]: Why is this separate? Does the setup differ between the 2? If so, let's create a separate module for this. If they don't differ, please create a nested `describe` block under the first `Mugshot integration` one.

**Comment by andrei-picus-hs** [comment]
Finished review.

File: tests/integration/test.js (hunk 3 above)
Comment (adela-istrate-hs) [inline tests/integration/test.js]: The first and second suite are testing Mugshot with the default rootDirectory. This function is needed to clean that.

File: tests/integration/test.js (hunk 6 above)
Comment (adela-istrate-hs) [inline tests/integration/test.js]: It differs because I have to give Mugshot the directory that `tmp` created. I'll create a new module.

**Comment by adela-istrate-hs** [comment]
@andrei-picus-hs  @valentin-radulescu-hs  can you review?

File: tests/integration/recurstive-root.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+const name = 'great',
+      ext = '.png',
+      extendedDir = 'visual-tests-extended/components/rectangle';
+
+const noDifferencesSelector = {
+  name: name,
+  selector: '#rectangle'
+};
+
+describe('Mugshot integration with recursive rootDirectory', function() {
```
Comment (andrei-picus-hs) [inline tests/integration/recurstive-root.js]:
```js
describe('Mocha integration', function() {
  describe('recursive rootDirectory', function() {
  });
});
```

**Comment by andrei-picus-hs** [comment]
Other than that LGTM.

**Comment by valentin-radulescu-hs** [comment]
LGTM

**Pull request merged**